### ISSUE TITLE
azurerm_storage_account docs seem contradictory

### ISSUE THREAD

**Issue by Netkracker** [hashicorp/terraform-provider-azurerm#25051]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.3

### AzureRM Provider Version

3.93.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
Current docs seem inconsistent regarding account_kind and account_tier for Premium Tier.
```

### Debug Output/Panic Output

```shell
none
```

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#account_kind

[account_kind](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#account_kind) - (Optional) Defines the Kind of account. Valid options are BlobStorage, BlockBlobStorage, FileStorage, Storage and StorageV2. Defaults to StorageV2.

[account_tier](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#account_tier) - (Required) Defines the Tier to use for this storage account. Valid options are Standard and Premium. For BlockBlobStorage and FileStorage accounts only Premium is valid. Changing this forces a new resource to be created.

NOTE:
Blobs with a tier of Premium are of account kind StorageV2.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/88662795/9c5cb79b-371b-489a-bed1-461053d9d9b2)

**Comment by magodo** [comment]
@Netkracker The doc LGTM, can you elaborate more on which part needs improvement?

**Comment by Netkracker** [comment]
account_tier states, account kinds “blockblobstoarge” and “Filestorage” are of account_tier “Premium”. But The Note states Blobs of account_tier “Premium” are account_kind “Storagev2”.

Currently when selecting account_tier “Premium” the only allowed option is “StorageV2” is this intended or is this a bug ?

**Comment by magodo** [comment]
@Netkracker 

> Currently when selecting account_tier “Premium” the only allowed option is “StorageV2”

How do you get this conclusion?

**Comment by Netkracker** [comment]
from this open issue and personal experience
https://github.com/hashicorp/terraform-provider-azurerm/issues/24782

when using "account_tier" = premium 
the only possible account_kind value allowed seems to be “StorageV2”, every other option fails with * `access_tier` is only available for accounts of kind: [BlobStorage StorageV2 FileStorage].

**Comment by magodo** [comment]
No.. The `account_tier = "Premium" doesn't limit your `account_kind` to `StorageV2` only. E.g. you can checkout [this acctest](https://github.com/hashicorp/terraform-provider-azurerm/blob/ed0f26c31d674ceef7344aa9da73b4305c9bf764/internal/services/storage/storage_account_resource_test.go#L1566-L1579), which has following config:

```
resource "azurerm_storage_account" "test" {
  name                = "unlikely23exst2acct%[3]s"
  resource_group_name = azurerm_resource_group.test.name

  location                 = azurerm_resource_group.test.location
  account_tier             = "Premium"
  account_replication_type = "LRS"
  account_kind             = "FileStorage"

  share_properties {
    retention_policy {
      days = 5
    }
  }

  lifecycle {
    ignore_changes = [share_properties.0.smb]
  }
}
``` 

Since this is a question rather than an issue in the Azure Provider I'm going to close this issue for the moment, but should you have further usage questions [I'd direct you towards the Community Discuss forum](https://discuss.hashicorp.com).
